### PR TITLE
Remove field from AggregateParams

### PR DESCRIPTION
**Pull request by rialg**
## Description
Refactor the class `AggregateParams` by removing the `public_partitions` field, thus making it serializable.

## How has this been tested?

```
$make test
==================================================================================================================== test session starts =====================================================================================================================
platform linux -- Python 3.9.10, pytest-7.1.2, pluggy-1.0.0
rootdir: /usr/local/google/home/grial/PipelineDP
plugins: timeout-2.1.0
collected 215 items                                                                                                                                                                                                                                          

tests/accumulator_test.py ...................                                                                                                                                                                                                          [  8%]
tests/budget_accounting_test.py ................                                                                                                                                                                                                       [ 16%]
tests/combiners_test.py .................................                                                                                                                                                                                              [ 31%]
tests/dp_computations_test.py ...............                                                                                                                                                                                                          [ 38%]
tests/dp_engine_test.py .........................s....                                                                                                                                                                                                 [ 52%]
tests/pipeline_backend_test.py ...............................................................                                                                                                                                                         [ 81%]
tests/private_beam_test.py .....................                                                                                                                                                                                                       [ 91%]
tests/private_spark_test.py ................                                                                                                                                                                                                           [ 99%]
tests/report_generator_test.py ..                                                                                                                                                                                                                      [100%]

====================================================================================================================== warnings summary ======================================================================================================================
../.local/lib/python3.9/site-packages/hdfs/config.py:15
  /usr/local/google/home/grial/.local/lib/python3.9/site-packages/hdfs/config.py:15: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    from imp import load_source

../.local/lib/python3.9/site-packages/apache_beam/testing/test_pipeline.py:36
  /usr/local/google/home/grial/.local/lib/python3.9/site-packages/apache_beam/testing/test_pipeline.py:36: PytestCollectionWarning: cannot collect test class 'TestPipeline' because it has a __init__ constructor (from: tests/private_beam_test.py)
    class TestPipeline(Pipeline):

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=================================================================================================== 214 passed, 1 skipped, 2 warnings in 560.84s (0:09:20) ===================================================================================================
```

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: pipeline_dp/dp_engine.py [hunk 1]
```diff
@@ -49,12 +49,13 @@ def __init__(self, budget_accountant: 'BudgetAccountant',
     def _add_report_stage(self, text):
         self._report_generators[-1].add_stage(text)
 
-    def aggregate(self, col, params: pipeline_dp.AggregateParams,
+    def aggregate(self, col, public_partitions, params: pipeline_dp.AggregateParams,
```
Comment (dvadym) [inline pipeline_dp/dp_engine.py:52]: `public_partitions` is optional argument. Could you please put them to the end of the list and make default value` = None`.

File: pipeline_dp/aggregate_params.py [hunk 2]
```diff
@@ -81,7 +78,6 @@ class AggregateParams:
     high: float = None  # deprecated
     min_value: float = None
     max_value: float = None
-    public_partitions: Any = None
```
Comment (dvadym) [inline pipeline_dp/aggregate_params.py:84]: This is a part of the public API, it's better to help the user to fix their old code. Could you please deprecate them the same as with `high` 3 lines above.

**Review by dvadym — CHANGES_REQUESTED** [review]
Thanks, it looks good! I left a few comments

Could you please also update the following example files:

[examples/quickstart.ipynb](https://github.com/OpenMined/PipelineDP/blob/main/examples/quickstart.ipynb)
[examples/restaurant_visits.ipynb](https://github.com/OpenMined/PipelineDP/blob/main/examples/restaurant_visits.ipynb)
[movie_view_ratings/run_all_frameworks.py](https://github.com/OpenMined/PipelineDP/blob/784961601d26afa0a08beedc7fe15587c3da9987/examples/movie_view_ratings/run_all_frameworks.py#L80)
[codelab/codelab_PipelineDP.ipynb](https://github.com/OpenMined/PipelineDP/blob/main/examples/codelab/codelab_PipelineDP.ipynb)

**Comment by dvadym** [comment]
Thank you! Changes LGTM! There are some format [errors](https://github.com/OpenMined/PipelineDP/runs/6349890896?check_suite_focus=true)

Could you please run `make format` from source root?

**Comment by rialg** [comment]
I am going ahead and update: `SumParams`, `VarianceParams`, `MeanParams`, `CountParams` and `PrivacyIdCountParams`

File: pipeline_dp/aggregate_params.py [hunk 3]
```diff
@@ -82,7 +79,7 @@ class AggregateParams:
...
+    public_partitions: Any = None  # deprecated
```
Comment (dvadym) [inline pipeline_dp/aggregate_params.py:82]: Could you please add add in __post_init(self) 5 lines below?

```
if self.public_partitions:
  raise ValueError(" AggregateParams.public_partitions is deprecated. Please use public_partitions argument in DPEngine.aggregate insead.")
```

File: pipeline_dp/aggregate_params.py [hunk 4]
```diff
@@ -82,7 +79,7 @@ class AggregateParams:
...
     noise_kind: NoiseKind = NoiseKind.LAPLACE
     custom_combiners: Iterable['CustomCombiner'] = None
```
Comment (dvadym) [inline pipeline_dp/aggregate_params.py:85]: Could you please deprecate (i.e. # deprecate and check in \_\_post_init\_\_)  `public_partitions` in `SumParams`, `CountParams`, `MeanParams`, `VarienceParams` as well?

**Review by dvadym — COMMENTED** [review]
Thank you for removing `public_partitions` in SumParams, MeanParams! Sry I realized that a few minor things missing (see my comments below)

**Review by dvadym — APPROVED** [review]
Thank you!

**Pull request merged**